### PR TITLE
Add techtree method in init, that return a dictionary with technology…

### PR DESCRIPTION
**Pull request by Systemac**
Add techtree method in init, that return a dictionary with technology, supply and facility necessary to obtain it.

Sorry for bad camelcase and variable'names...

### CODE REVIEW

File: ogame/__init__.py [hunk 1]
```diff
@@ -336,6 +337,27 @@ def inConstruction(self):
         else:
             return False
 
+    def techtree(self, id):
+        id = id(0)[0]
+        dico_tree = {}
+        r = self.session.get(
+            self.index_php + f'page=ajax&component=technologytree&ajax=1&technologyId={id}&tab=1'
+        ).text
+        bs3 = BeautifulSoup4(r)
```
Comment (PiecePaperCode) [inline ogame/__init__.py:346]: bs4

File: ogame/__init__.py [hunk 2]
```diff
@@ -894,7 +916,7 @@ def send_fleet(
 
         response = self.session.post(
             url=self.index_php + 'page=ingame&component=fleetdispatch'
-                '&action=sendFleet&ajax=1&asJson=1',
+                                 '&action=sendFleet&ajax=1&asJson=1',
```
Comment (PiecePaperCode) [inline ogame/__init__.py:919]: revert autostyling of your IDE

File: ogame/__init__.py [hunk 3]
```diff
@@ -336,6 +337,27 @@ def inConstruction(self):
...
             return False
 
+    def techtree(self, id):
+        id = id(0)[0]
+        dico_tree = {}
+        r = self.session.get(
+            self.index_php + f'page=ajax&component=technologytree&ajax=1&technologyId={id}&tab=1'
+        ).text
+        bs3 = BeautifulSoup4(r)
+        i = bs3.find('div', {'class': 'content technologytree'}).find_all('div', {'class': 'techWrapper'})
```
Comment (PiecePaperCode) [inline ogame/__init__.py:347]: Yeah it makes it hard to understand.
i is normaly a variable i assosiate with index and not an bs class

**Review by PiecePaperCode — CHANGES_REQUESTED** [review]

File: ogame/__init__.py (hunk 1 above)
Comment (Systemac) [reply ogame/__init__.py:346]: I'll change it

File: ogame/__init__.py (hunk 2 above)
Comment (Systemac) [reply ogame/__init__.py:919]: Oh sorry, it's Automatic...
I'll revert.

File: ogame/__init__.py (hunk 3 above)
Comment (Systemac) [reply ogame/__init__.py:347]: Oh ? Do you have an example ?

Comment (PiecePaperCode) [reply ogame/__init__.py:347]: call it what it is.
in this line your searching technologys.
tech, technology, techs, technologys, technology_tree, tree, tech_nodes, requirements, technology_requirement.

Comment (Systemac) [reply ogame/__init__.py:347]: so... it's just the link to grap the treetech, it's the same for the shipyard or another items..., just the id change (id from constants.py)
that's why I closed this request, I redid it a little more explicit ... Hoping that it is clearer ...

**Pull request closed without merging**